WordPress/Classes/ViewRelated/Post/PostSettingsViewController+FeaturedImageUpload.swift

```diff
@@ -0,0 +1,68 @@
+import Foundation
+import Photos
+
+extension PostSettingsViewController {
+
+    @objc func setFeaturedImage(asset: PHAsset) {
+        let media = MediaCoordinator.shared.addMedia(from: asset, to: self.apost)
+        setupObservingOf(media: media)
+    }
+
+    @objc func setFeaturedImage(media: Media) {
+        if media.hasRemote {
+            apost.featuredImage = media
+            return
+        }
+        MediaCoordinator.shared.retryMedia(media)
+        setupObservingOf(media: media)
+    }
+
+    @objc func removeMediaObserver() {
+        if let receipt = mediaObserverReceipt {
+            MediaCoordinator.shared.removeObserver(withUUID: receipt)
+            mediaObserverReceipt = nil
+        }
+    }
+
+    func setupObservingOf(media: Media) {
+        removeMediaObserver()
+        isUploadingMedia = true
+        mediaObserverReceipt = MediaCoordinator.shared.addObserver({ [weak self](media, state) in
+            self?.mediaObserver(media: media, state: state)
+        })
+        let progress = MediaCoordinator.shared.progress(for: media)
+        self.featuredImageProgress = progress
+    }
+
+    func mediaObserver(media: Media, state: MediaCoordinator.MediaState) {
+        switch state {
+        case .processing:
+            featuredImageProgress?.localizedDescription = NSLocalizedString("Preparing...", comment: "Label to show while converting and/or resizing media to send to server")
+        case .thumbnailReady:
+            if let url = media.absoluteThumbnailLocalURL, let data = try? Data(contentsOf: url) {
+                featuredImageProgress?.setUserInfoObject(UIImage(data: data), forKey: .WPProgressImageThumbnailKey)
+            }
+        case .uploading(let progress):
+            featuredImageProgress = progress
+            featuredImageProgress?.kind = .file
+            featuredImageProgress?.setUserInfoObject(Progress.FileOperationKind.copying, forKey: ProgressUserInfoKey.fileOperationKindKey)
+            featuredImageProgress?.localizedDescription = NSLocalizedString("Uploading...", comment: "Label to show while uploading media to server")
+            tableView.reloadData()
+        case .ended:
+            apost.featuredImage = media
+            isUploadingMedia = false
+            tableView.reloadData()
+        case .failed(let error):
+            DDLogError("Couldn't export image: /(error.localizedDescription)")
```

Comment:
Should be a `\`not `/`

Reply:
@frosty  So `Could not export image` ?

Should I change the message bellow to: `Could not upload featured image`

Reply:
Thanks!